python/tank/util/filesystem.py

```diff
@@ -343,3 +343,64 @@ def create_valid_filename(value):
         # so that we return a string
         u_src = value.decode("utf-8")
         return exp.sub("_", u_src).encode("utf-8")
+
+def get_permissions(path):
+    """
+    Retrieve the file system permissions for the file or folder in the
+    given path.
+
+    :param filename: Path to the file to be queried for permissions
+    :returns: permissions bits of the file 
+    :raises: OSError - if there was a problem retrieving permissions for the path
+    """
+    return stat.S_IMODE(os.lstat(path)[stat.ST_MODE])
+
+def delete_folder(path):
+    """
+    Deletes a folder and all of its contents recursively, even if it has read only items
+
+    :param path: File system path to location to the folder to be deleted
+    :returns: True if the folder and all of its contents were deleted without
```

Comment:
not totally sure about the return value here - looks like we are not using the return value, so maybe better to just leave it undefined for the time being? In the future, when we have a stronger use case for what we want returned (list of files? number of files? bool flag?), we can add it in.

That would make it consistent with https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/util/filesystem.py#L167 - 

i would even suggest renaming the method to `safe_delete_folder(path)` for consistency with `safe_delete_file(path)` - what do you think?